firefox_ui_tests/remote/security/test_ssl_status_after_restart.py

```diff
@@ -0,0 +1,127 @@
+          Services.obs.notifyObservers(cancelQuit, "quit-application-requested", null);
+        """)
+        self.restart()
+
+        i = 0
+        for item in self.test_data:
+            self.browser.tabbar.tabs[i].select()
+            self.verify_certificate_status(item)
+            i = i + 1
+
+    def verify_certificate_status(self, item):
+        url, identity, type = item['url'], item['identity'], item['type']
+
+        # Check the favicon
+        # TODO: find a better way to check, e.g., mozmill's isDisplayed
+        favicon_hidden = self.marionette.execute_script("""
+          return arguments[0].hasAttribute("hidden");
+        """, script_args=[self.browser.navbar.locationbar.favicon])
+        self.assertFalse(favicon_hidden)
+
+        self.identity_popup.box.click()
+        Wait(self.marionette).until(lambda _: self.identity_popup.is_open)
+
+        # Check the type shown on the idenity popup doorhanger
+        self.assertEqual(self.identity_popup.popup.get_attribute('className'),
+                         type,
+                         'Extended certificate is verified for ' + url)
+
+        # Check the identity label
+        self.assertEqual(self.identity_popup.organization_label.get_attribute('value'),
+                         identity,
+                         'Identity name is correct for ' + url)
+
+        # Get the information from the certificate
+        cert = self.browser.tabbar.selected_tab.certificate
+
+        # Open the Page Info window by clicking the More Information button
+        page_info = self.browser.open_page_info_window(
+            lambda _: self.identity_popup.more_info_button.click())
+
+        # Verify that the current panel is the security panel
+        self.assertEqual(page_info.deck.selected_panel, page_info.deck.security)
+
+        # Verify the domain listed on the security panel
+        # If this is a wildcard cert, use only the domain
+        if '*' in cert['commonName'][0]:
+            cert_name = self.security.get_domain_from_common_name(cert['commonName'])
+        else:
+            cert_name = cert['commonName']
+
+        self.assertIn(cert_name, page_info.deck.security.domain.get_attribute('value'),
+                      'Expected name found in certificate for ' + url)
+
+        # Verify the owner listed on the security panel
+        if identity != '':
+            owner = cert['organization']
+        else:
+            owner = page_info.get_property('securityNoOwner')
+
+        self.assertEqual(page_info.deck.security.owner.get_attribute('value'), owner,
+                         'Expected owner label found for ' + url)
+
+        # Verify the verifier listed on the security panel
+        self.assertEqual(page_info.deck.security.verifier.get_attribute('value'),
+                         cert['issuerOrganization'],
+                         'Verifier matches issuer of certificate for ' + url)
+        page_info.close()
```

Comment:
It's easier to see what's going on with this in place, but it's true, it's not apparently necessary here. I've added the window.close that the page info unit test uses to tearDown.